Source/ACE/Entity/WorldObject.cs

```diff
@@ -606,21 +638,91 @@ protected WorldObject(ObjectGuid guid, AceObject aceObject)
             PaletteGuid = aceObject.PaletteId;
         }
 
-        public void SetCombatMode(CombatMode newCombatMode)
+        public void SetCombatMode(CombatMode newCombatMode, Player player)
```

Comment:
This confuses me -- Why is SetCombatMode a function of WorldObject and not player?  Additionally, why is Player passed into it?  What context is it called from?  (e.g. is it safe to access the player from this context? -- probably not).

This smells funny to me, and I would like to understand it more.